src/main/java/org/deepsymmetry/beatlink/data/OpusProvider.java

```diff
@@ -51,98 +59,135 @@ public boolean isRunning() {
 
     /**
-     * Returns opus reported slot for archive 2
-     * @return the TrackSourceSlot for this archive
+     * Attach a metadata archive to supply information for the media mounted a USB slot of the Opus Quad.
+     * This is a convenience method to be backwards compatible for BeatLinkTrigger, but this should probably go.
+     *
+     * @param archiveFile the metadata archive that can provide metadata for tracks playing from the specified USB slot, or
+     *                {@code null} to stop providing metadata for that slot
+     * @param slot which TrackSourceSlot slot we are to provide metadata for. They are mapped to the USB slots
+     *
+     * @throws java.io.IOException if there is a problem attaching the archive
+     * @throws IllegalArgumentException if a slot other than the SD or USB slot is specified
      */
-    public AtomicReference<CdjStatus.TrackSourceSlot> getOpusReportedDataSlot2() {
-        return opusReportedDataSlot2;
+    @API(status = API.Status.STABLE)
+    public synchronized void attachMetadataArchive(File archiveFile, CdjStatus.TrackSourceSlot slot) throws IOException {
+        switch(slot){
+            case USB_SLOT:
+                attachMetadataArchive(archiveFile, 1);
+                break;
+            case SD_SLOT:
+                attachMetadataArchive(archiveFile, 2);
+                break;
+            case CD_SLOT:
+                attachMetadataArchive(archiveFile, 3);
+        }
     }
 
     /**
      * Attach a metadata archive to supply information for the media mounted a USB slot of the Opus Quad.
      * This must be a file created using {@link org.deepsymmetry.cratedigger.Archivist#createArchive(Database, File)}
      * from that media.
      *
-     * @param archive the metadata archive that can provide metadata for tracks playing from the specified USB slot, or
+     * @param archiveFile the metadata archive that can provide metadata for tracks playing from the specified USB slot, or
      *                {@code null} to stop providing metadata for that slot
-     * @param slot which USB slot we are to provide metadata for (we follow the XDJ-XZ convention of using
-     *            {@link CdjStatus.TrackSourceSlot#SD_SLOT} to represent USB 1, and
-     *            {@link CdjStatus.TrackSourceSlot#USB_SLOT} to represent USB 2)
+     * @param usbSlotNumber which USB slot we are to provide metadata for. Acceptable values are 1, 2 and 3.
      *
      * @throws java.io.IOException if there is a problem attaching the archive
      * @throws IllegalArgumentException if a slot other than the SD or USB slot is specified
      */
     @API(status = API.Status.STABLE)
-    public synchronized void attachMetadataArchive(File archive, CdjStatus.TrackSourceSlot slot) throws IOException {
+    public synchronized void attachMetadataArchive(File archiveFile, int usbSlotNumber) throws IOException {
```

Comment:
At this point it probably makes sense to use a map from slot number to archive references to store these.

Reply:
Yup, would be better. Adjusting!

Reply:
Oh, you even already had one! So we don’t need the individual instance variables any more.

Reply:
Yes, getting rid of these simplifies several parts of the code.

Reply:
Please don’t do anything, I am in the middle of a major refactor.